Source/WeatherContent/WeatherViewController.swift

```diff
@@ -0,0 +1,149 @@
+//  Copyright © 2016 Gavan Chan. All rights reserved.
+
+import RxCocoa
+import RxSwift
+import UIKit
+
+final class WeatherViewController: UIViewController {
+
+  private var currentModel: Weather? = nil
+
+  private let disposables: CompositeDisposable = CompositeDisposable()
+
+  private var weatherTransitioningDisposableKey: CompositeDisposable.DisposeKey? = nil
+
+  private var contentProducer: Observable<Weather?> = .empty() {
+    didSet {
+      if let key = weatherTransitioningDisposableKey { disposables.remove(for: key) }
+
+      let disposable: Disposable = transitions
+        .observeOn(MainScheduler.asyncInstance)
+        .subscribe { [weak self] in self?.perform(transition: $0) }
+
+      weatherTransitioningDisposableKey = disposables.insert(disposable)
+    }
+  }
+
+  private var transitions: Observable<Transition<Weather>> {
```

Comment:
As per a commit description, an issue arises from starting this observable when the view reappears on the screen where instead of clearing its current content and displaying a state which represents the view having "no content", the most recent content remains on the screen, even though `nilBlock` in the test observable is prefixing its contents with `nil`.